File: fix-zindex.js
const fs = require('fs');

// Read the templates file
const templateFile = './src/utils/templates.js';
let content = fs.readFileSync(templateFile, 'utf8');

// Replace all zIndex: 0 with proper values based on shape type
// Text elements should have highest zIndex
content = content.replace(/type: 'text',\n([\s\S]*?)zIndex: 0/g, (match) => {
  return match.replace('zIndex: 0', 'zIndex: 10');
});

// Circles (usually decorative/icons) - medium zIndex
content = content.replace(/type: 'circle',\n([\s\S]*?)zIndex: 0/g, (match) => {
  return match.replace('zIndex: 0', 'zIndex: 5');
});

// Rectangles - check if it's a background or content
// We'll do this in multiple passes:
// 1. Large backgrounds (width > 8000 or height > 8000) - zIndex: 1
// 2. Medium containers - zIndex: 2  
// 3. Content boxes - zIndex: 3

// First pass: find rectangles
const lines = content.split('\n');
let inShape = false;
let shapeType = null;
let shapeWidth = 0;
let shapeHeight = 0;
let shapeZIndexLine = -1;
let result = [];

for (let i = 0; i < lines.length; i++) {
  const line = lines[i];
  
  if (line.includes("type: 'rectangle',") || line.includes('type: "rectangle",')) {
    inShape = true;
    shapeType = 'rectangle';
    shapeWidth = 0;
    shapeHeight = 0;
  } else if (line.includes("type: 'text',") || line.includes('type: "text",')) {
    inShape = true;
    shapeType = 'text';
  } else if (line.includes("type: 'circle',") || line.includes('type: "circle",')) {
    inShape = true;
    shapeType = 'circle';
  }
  
  if (inShape && shapeType === 'rectangle') {
    const widthMatch = line.match(/width:\s*(\d+)/);
    const heightMatch = line.match(/height:\s*(\d+)/);
    if (widthMatch) shapeWidth = parseInt(widthMatch[1]);
    if (heightMatch) shapeHeight = parseInt(heightMatch[1]);
    
    if (line.includes('zIndex:') && line.includes('0')) {
      // Determine zIndex based on size
      let newZIndex = 3; // Default for content
      if (shapeWidth >= 8000 || shapeHeight >= 8000) {
        newZIndex = 1; // Background
      } else if (shapeWidth >= 5000 || shapeHeight >= 5000) {
        newZIndex = 2; // Large container
      }
      lines[i] = line.replace(/zIndex:\s*0/, `zIndex: ${newZIndex}`);
      inShape = false;
    }
  } else if (inShape && shapeType === 'text') {
    if (line.includes('zIndex:') && line.includes('0')) {
      lines[i] = line.replace(/zIndex:\s*0/, 'zIndex: 10');
      inShape = false;
    }
  } else if (inShape && shapeType === 'circle') {
    if (line.includes('zIndex:') && line.includes('0')) {
      lines[i] = line.replace(/zIndex:\s*0/, 'zIndex: 5');
      inShape = false;
    }
  }
  
  if (line.includes('}') && line.trim() === '},' || line.trim() === '}') {
    inShape = false;
  }
}

content = lines.join('\n');

// Write back
fs.writeFileSync(templateFile, content, 'utf8');
console.log('Fixed zIndex values in templates.js');

File: fix-zindex.py
#!/usr/bin/env python3
import re

# Read the file
with open('src/utils/templates.js', 'r') as f:
    content = f.read()

# Split into shapes
shapes = content.split('    {')
fixed_shapes = [shapes[0]]  # Keep the beginning

for shape in shapes[1:]:
    # Check if this is a text shape
    if "type: 'text'" in shape or 'type: "text"' in shape:
        # Replace zIndex: 0 with zIndex: 10 for text
        shape = re.sub(r'zIndex: 0', 'zIndex: 10', shape)
    
    # Check if this is a circle
    elif "type: 'circle'" in shape or 'type: "circle"' in shape:
        # Replace zIndex: 0 with zIndex: 5 for circles
        shape = re.sub(r'zIndex: 0', 'zIndex: 5', shape)
    
    # Check if this is a rectangle - need to be smarter
    elif "type: 'rectangle'" in shape or 'type: "rectangle"' in shape:
        # Extract width and height
        width_match = re.search(r'width: (\d+)', shape)
        height_match = re.search(r'height: (\d+)', shape)
        
        if width_match and height_match:
            width = int(width_match.group(1))
            height = int(height_match.group(1))
            
            # Determine zIndex based on size
            if width >= 10000 or height >= 10000:
                # Very large background
                shape = re.sub(r'zIndex: 0', 'zIndex: 1', shape)
            elif width >= 8000 or height >= 8000:
                # Large background
                shape = re.sub(r'zIndex: 0', 'zIndex: 1', shape)
            elif width >= 5000 or height >= 5000:
                # Medium container
                shape = re.sub(r'zIndex: 0', 'zIndex: 2', shape)
            else:
                # Content box
                shape = re.sub(r'zIndex: 0', 'zIndex: 3', shape)
    
    fixed_shapes.append(shape)

# Join back
content = '    {'.join(fixed_shapes)

# Write back
with open('src/utils/templates.js', 'w') as f:
    f.write(content)

print("Fixed zIndex values!")
print("Text shapes: zIndex 10")
print("Circle shapes: zIndex 5")
print("Rectangle backgrounds (>=8000): zIndex 1")
print("Rectangle containers (>=5000): zIndex 2")
print("Rectangle content: zIndex 3")

